src/main/webapp/app/exercises/programming/manage/instructions-editor/analysis/programming-exercise-instruction-analysis.service.ts

```diff
@@ -113,19 +107,19 @@ export class ProgrammingExerciseInstructionAnalysisService {
      * @param analysis arbitrary number of analysis objects to be merged into one.
      */
     private mergeAnalysis = (...analysis: Array<AnalysisItem[]>) => {
-        return compose(
-            reduce((acc, [lineNumber, values, issueType]) => {
-                const lineNumberValues = acc[lineNumber];
-                const issueValues = lineNumberValues ? lineNumberValues[issueType] || [] : [];
-                return { ...acc, [lineNumber]: { ...lineNumberValues, [issueType]: [...issueValues, ...values] } };
-            }, {}),
-            map(([lineNumber, values, issueType]: AnalysisItem) => [
+        const reducer = (acc: ProblemStatementAnalysis, [lineNumber, values, issueType]: AnalysisItem) => {
+            const lineNumberValues = acc[lineNumber];
+            const issueValues = lineNumberValues ? lineNumberValues[issueType] || [] : [];
+            return { ...acc, [lineNumber]: { ...lineNumberValues, [issueType]: [...issueValues, ...values] } };
+        };
+        return analysis
+            .flat()
```

Comment:
Why the additional flat here?

Reply:
Because that was the first operation in the old code as well (old line 127). `analysis` is an Array of `AnalysisItem[]`.

Reply:
Ah, it decided to hide below 😂 Makes sense.